test/functional/lua/system_spec.lua

```diff
@@ -120,4 +119,31 @@ describe('vim.system', function()
       system_sync({ './test' })
     end)
   end
+
+  it('always captures all content of stdout/stderr #30846', function()
+    eq(
+      0,
+      exec_lua(function()
+        local done = 0
+        local fail = 0
+        for _ = 1, 200 do
+          vim.system(
+            { 'git', 'show', ':0:test/functional/plugin/lsp_spec.lua' },
```

Comment:
Do we need to skip this if `git` is not available (I think that might be the case for Debian's package build system)?

    skip(fn.executable('git') == 0, 'missing "git" command')

https://github.com/neovim/neovim/blob/ae93c7f369a174f3d738ab55030de2c9dfc10c57/test/functional/core/channels_spec.lua#L323